include/nbl/video/SPhysicalDeviceFeatures.h

```diff
@@ -0,0 +1,553 @@
+    //VkBool32           primitivesGeneratedQueryWithRasterizerDiscard;
+    //VkBool32           primitivesGeneratedQueryWithNonZeroStreams;
+
+    /* PrimitiveTopologyListRestartFeaturesEXT *//* VK_EXT_primitive_topology_list_restart */
+    //VkBool32           primitiveTopologyListRestart;
+    //VkBool32           primitiveTopologyPatchListRestart;
+
+    /* PrivateDataFeatures *//* VK_EXT_private_data *//* MOVED TO Vulkan 1.3 Core */
+
+    /* ProvokingVertexFeaturesEXT *//* VK_EXT_provoking_vertex */
+    //VkBool32           provokingVertexLast;
+    //VkBool32           transformFeedbackPreservesProvokingVertex;
+
+    /* Robustness2FeaturesEXT *//* VK_EXT_robustness2 */
+    //VkBool32           robustBufferAccess2;
+    //VkBool32           robustImageAccess2;
+    //VkBool32           nullDescriptor;
+    
+    /* ScalarBlockLayoutFeaturesEXT *//* VK_EXT_scalar_block_layout *//* MOVED TO Vulkan 1.2 Core */
+    
+    /* ShaderAtomicFloatFeaturesEXT *//* VK_EXT_shader_atomic_float */
+    //VkBool32           shaderBufferFloat32Atomics;
+    //VkBool32           shaderBufferFloat32AtomicAdd;
+    //VkBool32           shaderBufferFloat64Atomics;
+    //VkBool32           shaderBufferFloat64AtomicAdd;
+    //VkBool32           shaderSharedFloat32Atomics;
+    //VkBool32           shaderSharedFloat32AtomicAdd;
+    //VkBool32           shaderSharedFloat64Atomics;
+    //VkBool32           shaderSharedFloat64AtomicAdd;
+    //VkBool32           shaderImageFloat32Atomics;
+    //VkBool32           shaderImageFloat32AtomicAdd;
+    //VkBool32           sparseImageFloat32Atomics;
+    //VkBool32           sparseImageFloat32AtomicAdd;
+
+    /* ShaderAtomicFloat2FeaturesEXT *//* VK_EXT_shader_atomic_float2 */
+    //VkBool32           shaderBufferFloat16Atomics;
+    //VkBool32           shaderBufferFloat16AtomicAdd;
+    //VkBool32           shaderBufferFloat16AtomicMinMax;
+    //VkBool32           shaderBufferFloat32AtomicMinMax;
+    //VkBool32           shaderBufferFloat64AtomicMinMax;
+    //VkBool32           shaderSharedFloat16Atomics;
+    //VkBool32           shaderSharedFloat16AtomicAdd;
+    //VkBool32           shaderSharedFloat16AtomicMinMax;
+    //VkBool32           shaderSharedFloat32AtomicMinMax;
+    //VkBool32           shaderSharedFloat64AtomicMinMax;
+    //VkBool32           shaderImageFloat32AtomicMinMax;
+    //VkBool32           sparseImageFloat32AtomicMinMax;
+    
+    /* DemoteToHelperInvocationFeaturesEXT *//* VK_EXT_shader_demote_to_helper_invocation *//* MOVED TO Vulkan 1.3 Core */
+
+    /* ShaderImageAtomicInt64FeaturesEXT *//* VK_EXT_shader_image_atomic_int64 */
+    //VkBool32           shaderImageInt64Atomics;
+    //VkBool32           sparseImageInt64Atomics;
+
+    /* SubgroupSizeControlFeaturesEXT *//* VK_EXT_subgroup_size_control *//* MOVED TO Vulkan 1.3 Core */
+
+    /* TexelBufferAlignmentFeaturesEXT *//* VK_EXT_texel_buffer_alignment */
+    //VkBool32           texelBufferAlignment;
```

Comment:
promoted to 1.3 core, its only a bool XD useless to expose

Reply:
we'll "fake" it as if the implementation always supports the extension

Reply:
see https://github.com/Devsh-Graphics-Programming/Nabla/pull/357/files#r895536075

basically don't expose, always enable if we can